sidechain/src/lib.rs

```diff
@@ -118,15 +119,19 @@ pub mod pallet {
 			if block_number > Self::add_to_block_number(latest_block_number, lenience)? {
 				// Block is far too early and hence refused.
 				return Err(<Error<T>>::BlockNumberTooHigh.into())
-			} else if block_number > Self::add_to_block_number(latest_block_number, 1)? {
+			} else if block_number >
```

Comment:
Since the validateers choose the finality candidates, and that is one per parentchain block, I don't think we need this lenience and block queue any more, do we? @clangenb

Reply:
Emphasizing this again - do we still need this complicated logic in the pallet? I believe we can remove the block queue fully.